src/storage/src/storage/signed_url.rs

```diff
@@ -0,0 +1,367 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{error::SigningError, signed_url::UrlStyle};
+use auth::signer::Signer;
+use std::collections::BTreeMap;
+use tokio::time::Duration;
+
+/// Creates signed URLs.
+///
+/// This builder allows you to generate signed URLs for Google Cloud Storage objects and buckets.
+/// Signed URLs provide a way to give time-limited read or write access to specific resources
+/// without sharing your credentials.
+///
+/// # Example
+///
+/// ```no_run
+/// use google_cloud_storage::builder::storage::SignedUrlBuilder;
+/// use std::time::Duration;
+/// # use auth::signer::Signer;
+///
+/// # async fn run(signer: &Signer) -> Result<(), Box<dyn std::error::Error>> {
+/// let url = SignedUrlBuilder::for_object("my-bucket", "my-object.txt")
+///     .with_method(http::Method::GET)
+///     .with_expiration(Duration::from_secs(3600)) // 1 hour
+///     .sign_with(signer)
+///     .await?;
+///
+/// println!("Signed URL: {}", url);
+/// # Ok(())
+/// # }
+/// ```
```

Comment:
I can't remember if this is done elsewhere, but we probably want to link to the GCS docs for Signed URLs https://docs.cloud.google.com/storage/docs/access-control/signed-urls